dashboard/app/controllers/followers_controller.rb

```diff
@@ -5,74 +5,32 @@
 
 class FollowersController < ApplicationController
   before_action :authenticate_user!, except: [:student_user_new, :student_register]
-  before_action :load_section, only: [:create, :create_sync, :student_user_new, :student_register]
-
-  # join a section as a logged in student
-  def create
-    @section.add_student current_user
-
-    redirect_to redirect_url, notice: I18n.t('follower.added_teacher', name: @section.teacher.name)
-  end
-
-  # Remove enrollment in a section (as a student in the section).
-  def remove
-    @section = Section.find_by_code params[:section_code]
-    if @section
-      f = Follower.where(section: @section.id, student_user_id: current_user.id).first
-    end
-
-    unless @section && f
-      # TODO(asher): Change the alert message to section.
-      redirect_to root_path, alert: t(
-        'follower.error.section_not_found',
-        section_code: params[:section_code]
-      )
-      return
-    end
-
-    @teacher = @section.user
-
-    authorize! :destroy, f
-    f.delete
-    # Though in theory required, we are missing an email address for many teachers.
-    if @teacher && @teacher.email.present?
-      FollowerMailer.student_disassociated_notify_teacher(@teacher, current_user).deliver_now
-    end
-    teacher_name = @teacher ? @teacher.name : I18n.t('user.deleted_user')
-    redirect_to(
-      root_path,
-      notice: t(
-        'teacher.student_teacher_disassociated',
-        teacher_name: teacher_name,
-        section_code: params[:section_code]
-      )
-    )
-  end
+  before_action :load_section, only: [:student_user_new, :student_register]
 
   # GET /join/XXXXXX
   def student_user_new
     # Though downstream validations would raise an exception, we redirect to the admin directory to
     # improve user experience.
-    if current_user && current_user.admin?
+    if current_user&.admin?
       redirect_to admin_directory_path
       return
     end
 
-    if @section && @section.provider_managed?
+    # Redirect and provide an error for provider-managed sections.
+    if @section&.provider_managed?
       provider = I18n.t(@section.login_type, scope: 'section.type')
       redirect_to root_path, alert: I18n.t('follower.error.provider_managed_section', provider: provider)
       return
     end
 
-    @user = current_user || User.new
-
-    # if this is a picture or word section, redirect to the section login page so that the student
-    # does not have to type in the full URL
-    if @section && [Section::LOGIN_TYPE_PICTURE, Section::LOGIN_TYPE_WORD].include?(@section.login_type)
+    # If this is a picture or word section, redirect to the section login page so that the student
+    # does not have to type in the full URL.
+    if [Section::LOGIN_TYPE_PICTURE, Section::LOGIN_TYPE_WORD].include?(@section&.login_type)
```

Comment:
I'm not sure if this is a recent regression and/or being fixed in this PR, but word / picture section login does not look so good on production right now.

On production today, when I create a picture section, the Manage Students tab says for students to go to /join and enter the section code.

when I sign out and then go to /join and then put in the section code for a picture section, I am prompted to enter my name, email and password, which seems wrong.

If I navigate directly to /join/XXXXXX, I get redirected to /sections/XXXXXX, which seems right, however we don't include the link to /join/XXXXXX anywhere so the student will likely never find this.

perhaps a good starting point is to add a unit test that POSTing `XXXXXX` (picture section) to /join also redirects to /sections/XXXXXX, and then making sure that test passes?

Reply:
great catch! i would guess that's a recent regression from my previous refactor, but can't say for sure. i'll fix that in this PR